src/util/Permissions.js

```diff
@@ -32,12 +32,26 @@ class Permissions {
 
   /**
    * Gets all given permissions that are missing from the bitfield.
-   * @param {PermissionResolvable[]} permissions Permissions to check for
+   * @param {PermissionResolvable|PermissionResolvable[]} permissions Permission(s) to check for
    * @param {boolean} [checkAdmin=true] Whether to allow the administrator permission to override
    * @returns {PermissionResolvable[]}
    */
   missing(permissions, checkAdmin = true) {
-    return permissions.filter(p => !this.has(p, checkAdmin));
+    if (permissions instanceof Array) return permissions.filter(p => !this.has(p, checkAdmin));
+    return this.missing(new Permissions(permissions).toArray(checkAdmin), checkAdmin);
```

Comment:
Should construct `this.constructor` instead of `Permissions` in order to support method extends/overwrites.

Reply:
Thanks for noting this; I mainly use TypeScript so I'm not so familiar with OOP in plain JS

Reply:
Should `new` be used with `this.constructor`?

Reply:
yes